state/cleanup.go

```diff
@@ -177,12 +177,9 @@ func (st *State) cleanupModelsForDyingController() (err error) {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	for _, env := range models {
-
-		if env.Life() == Alive {
-			if err := env.Destroy(); err != nil {
-				return errors.Trace(err)
-			}
+	for _, model := range models {
```

Comment:
No: like every other Destroy method, multiple calls are safe and do not
error out. ErrModelNotAlive is an internal signal that's caught and turned
into ErrNoOperations in the buildTxn loop.

On Friday, July 1, 2016, John Arbash Meinel <notifications@github.com>
wrote:

> In state/cleanup.go
> <https://github.com/juju/juju/pull/5754#discussion_r69324718>:
>
> > @@ -177,12 +177,9 @@ func (st *State) cleanupModelsForDyingController() (err error) {
> >  	if err != nil {
> >  		return errors.Trace(err)
> >  	}
> > -	for _, env := range models {
> > -
> > -		if env.Life() == Alive {
> > -			if err := env.Destroy(); err != nil {
> > -				return errors.Trace(err)
> > -			}
> > +	for _, model := range models {
>
> Looking at DestroyOps() won't this cause an early failure with
> "errModelNotAlive" in the case that we run this and one of the models is
> already dying but not dead yet? eg:
> juju add-model foo
> juju add-model bar
> juju destroy-model foo
> juju destroy-controller
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju/juju/pull/5754/files/846d96b58d97f9a200daf3a81e71240242c20b30#r69324718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAWtbjlga1dmIJolSXluZ1JEL8RyBaCfks5qRT_2gaJpZM4JDOzu>
> .
>